iceoryx_hoofs/test/moduletests/test_relative_pointer.cpp

```diff
@@ -171,67 +199,75 @@ TYPED_TEST(RelativePointer_test, ConstrTests)
 TYPED_TEST(RelativePointer_test, AssignmentOperatorTests)
 {
     ::testing::Test::RecordProperty("TEST_ID", "cd0c4a6a-7779-4dc3-97dc-58ef40a58715");
-    auto ptr0 = this->partitionPtr(0);
-    auto ptr1 = this->partitionPtr(1);
+    auto* ptr0 = this->partitionPtr(0U);
+    auto* ptr1 = this->partitionPtr(1U);
 
-    EXPECT_EQ(BaseRelativePointer::registerPtr(1, ptr0, SHARED_MEMORY_SIZE), true);
-    EXPECT_EQ(BaseRelativePointer::registerPtr(2, ptr1, SHARED_MEMORY_SIZE), true);
+    EXPECT_EQ(BaseRelativePointer::registerPtr(iox::rp::BaseRelativePointer::id_t{1U}, ptr0, SHARED_MEMORY_SIZE), true);
+    EXPECT_EQ(BaseRelativePointer::registerPtr(iox::rp::BaseRelativePointer::id_t{2U}, ptr1, SHARED_MEMORY_SIZE), true);
 
     {
         RelativePointer<TypeParam> rp;
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr0);
+        auto* typedPtr = static_cast<TypeParam*>(static_cast<void*>(ptr0));
         rp = typedPtr;
-        EXPECT_EQ(rp.getOffset(), 0);
-        EXPECT_EQ(rp.getId(), 1);
+        EXPECT_EQ(rp.getOffset(), 0U);
+        EXPECT_EQ(rp.getId(), 1U);
         EXPECT_NE(rp, nullptr);
     }
 
     {
-        auto offset = SHARED_MEMORY_SIZE / 2;
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
+        auto offset = SHARED_MEMORY_SIZE / 2U;
+        // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+        auto* typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
         RelativePointer<TypeParam> rp;
         rp = typedPtr;
         EXPECT_EQ(rp.getOffset(), offset);
-        EXPECT_EQ(rp.getId(), 1);
+        EXPECT_EQ(rp.getId(), 1U);
         EXPECT_NE(rp, nullptr);
     }
 
     {
-        auto offset = SHARED_MEMORY_SIZE - 1;
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
+        auto offset = SHARED_MEMORY_SIZE - 1U;
+        // NOLINTJUSTIFICATION Pointer arithmetic needed for tests
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic, cppcoreguidelines-pro-type-reinterpret-cast)
+        auto* typedPtr = reinterpret_cast<TypeParam*>(ptr0 + offset);
         RelativePointer<TypeParam> rp;
         rp = typedPtr;
         EXPECT_EQ(rp.getOffset(), offset);
-        EXPECT_EQ(rp.getId(), 1);
+        EXPECT_EQ(rp.getId(), 1U);
         EXPECT_NE(rp, nullptr);
     }
 
     {
         RelativePointer<TypeParam> rp;
-        auto typedPtr = reinterpret_cast<TypeParam*>(ptr1);
+        auto* typedPtr = static_cast<TypeParam*>(static_cast<void*>(ptr1));
```

Comment:
please use reinterpret_cast

Reply:
`static_cast` has several advantages e.g. it can be used during compile-time, see [here](https://github.com/eclipse-iceoryx/iceoryx/pull/1521#discussion_r936463947).